yt/frontends/parthenon/io.py

```diff
@@ -52,7 +52,10 @@ def _read_fluid_selection(self, chunks, selector, fields, size):
                 for gs in grid_sequences(chunk.objs):
                     start = gs[0].id - gs[0]._id_offset
                     end = gs[-1].id - gs[-1]._id_offset + 1
-                    data = ds[start:end, fdi, :, :, :].transpose()
+                    if len(ds.shape) == 4:
+                        data = ds[start:end, :, :, :].transpose()
+                    else:
+                        data = ds[start:end, fdi, :, :, :].transpose()
```

Comment:
Parthenon's python reader can always split a variable into a flat index space, so I *think* this is now generic.